src/App.js

```diff
@@ -1,19 +1,143 @@
 import React from 'react';
 import './App.css';
+import { TodoList } from './components/TodoList';
 
+import todos from './api/todos';
 import users from './api/users';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+    user: '',
+    title: '',
+    isCorrectTitle: false,
+    isCorrectUserName: false,
+  }
+
+  addTodo = () => {
+    this.setState(state => ({
+      todos: [...state.todos, {
+        userId: users.find(user => user.name === state.user).id,
+        id: state.todos[state.todos.length - 1].id + 1,
+        title: state.title,
+        completed: false,
+        user: users.find(user => user.name === state.user),
+      }],
+    }));
+  }
+
+  hendlerFormChange = (event) => {
+    const { user, title } = this.state;
+
+    event.preventDefault();
+    this.setState(state => ({
+      isCorrectUserName: !user,
+    }));
+
+    if (title.length < 3) {
+      this.setState({
```

Comment:
We can add empty todo (using spaces)
![empty](https://user-images.githubusercontent.com/66570142/119940655-2e70a300-bf98-11eb-9e87-5f3156258067.png)